Servers/WindowServer/WSMenu.h

```diff
@@ -94,5 +100,7 @@ class WSMenu final : public CObject {
     WSMenuItem* m_hovered_item { nullptr };
     NonnullOwnPtrVector<WSMenuItem> m_items;
     RefPtr<WSWindow> m_menu_window;
+    WSWindow* m_window_menu_of { nullptr };
```

Comment:
Let's use a `WeakPtr<WSWindow>` for this to prevent ending up with a stale pointer in case the window goes away for some reason.